R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,196 @@
+        L154.NBS_trn_share_province
+
+      # Match these percentages into a table of all transportation technologies that will be written out
+      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
+        repeat_add_columns(tibble::tibble(province = gcamchina.PROVINCES)) ->
+      L154.pct_province_trn_m_sz_tech_F_Yh
+
+      # TODO: Tibet
+      #L154.pct_province_trn_m_sz_tech_F_Yh <- subset( L154.pct_province_trn_m_sz_tech_F_Yh, province != "XZ" )
+
+      L154.pct_province_trn_m_sz_tech_F_Yh %>%
+        filter(year %in% HISTORICAL_YEARS) %>%
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
+        left_join(L154.NBS_trn_share_province, by = c("province", "EBProcess", "EBMaterial", "year")) %>%
+        replace_na(list(value_share = 0)) %>%
+        mutate(value = NULL) ->
+        L154.pct_province_trn_m_sz_tech_F_Yh
+
+      #Now, the full CHINA tran UCD database can be apportioned to the provinces
+      # Creating the first of the three output tables
+      L154.pct_province_trn_m_sz_tech_F_Yh%>%
+        left_join_error_no_match(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh, by = c("year", "UCD_sector", "mode", "size.class", "UCD_technology", "UCD_fuel","fuel")) %>%
+        mutate(value = value * value_share) %>% # Allocating across the provinces
+        select(province, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel, year, value) ->
+        L154.in_EJ_province_trn_m_sz_tech_F
+
+      # As a final step, aggregate by fuel and name the sector
+      # This creates the second of three output tables
+      L154.in_EJ_province_trn_m_sz_tech_F %>%
+        group_by(province, fuel, year) %>%
+        summarise(value = sum(value)) %>%
+        ungroup() %>%
+        mutate(sector = "transportation") %>% # Adding a column named "sector" with "transportation" as the entries
+        select(province, sector, fuel, year, value) ->
+        L154.in_EJ_province_trn_F
+
+      # Apportion non-motorized energy consumption to provinces on the basis of population
+      # First we will create the province shares based on population
+      L100.Pop_thous_province %>%
+        complete( province, year = 1971:2100) %>%
+        group_by(province) %>%
+        mutate(pop = approx_fun(year, pop, rule = 2)) %>%
+        ungroup() %>%
+        group_by(year) %>%
+        summarise(value_national = sum(pop, na.rm =T)) ->
+        Pop_national
+
+      L100.Pop_thous_province %>%
+        complete( province, year = 1971:2100) %>%
+        group_by(province) %>%
+        mutate(pop = approx_fun(year, pop, rule = 2)) %>%
+        ungroup() %>%
+        left_join_error_no_match(Pop_national, by = "year") %>%
+        mutate(value_share = pop / value_national) %>% # Creating province share based on population
+        select(province, year, value_share) ->
+        Pop_province_share
+
+
+        # Now we can use these shares to allocate the national data across the provinces
+        L154.out_mpkm_R_trn_nonmotor_Yh %>%
```

Comment:
Can you do the same for the other output? my comments above - no need to repeat the provinces for the national tibble